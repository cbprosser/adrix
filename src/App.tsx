import { useMemo } from 'react';
import { Route, Routes } from 'react-router-dom';
import { sitemap } from './sitemap';

export const App = () => {
  const routes = useMemo(
    () =>
      (Object.keys(sitemap) as (keyof typeof sitemap)[]).reduce(
        (accumulator, key) => {
          const { Component, urls } = sitemap[key];
          return [
            ...accumulator,
            ...urls.map((url) => (
              <Route key={`${key}${url}`} path={url} element={<Component />} />
            )),
          ];
        },
        [] as JSX.Element[],
      ),
    [],
  );
  return (
    <div className="App">
      <Routes>{routes}</Routes>
    </div>
  );
};
