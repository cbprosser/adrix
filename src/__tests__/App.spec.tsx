import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { App as ImportedApp } from '../App';
import { mockAComponent } from './testUtils';

jest.mock('../components/Pages', () => ({
  Home: (props: any) => mockAComponent('Home')(props),
  Results: (props: any) => mockAComponent('Results')(props),
}));

const App = () => (
  <BrowserRouter>
    <ImportedApp />
  </BrowserRouter>
);

describe('App Suite', () => {
  it('Should match the snapshot', () => {
    const view = render(<App />);

    expect(view.asFragment()).toMatchSnapshot();
  });
});
