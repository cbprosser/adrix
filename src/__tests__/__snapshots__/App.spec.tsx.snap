// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Suite Should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-fixed css-1-MuiContainer-root"
      >
        <div
          id="Home"
        >
          {}
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
